File: cellab_devkit-1.0 (1)/ruledefs/fredmem.js
    /*

        The Fredkin rule with the seven extra bits used as memory

    */

    rule.worldtype = 1;          // 2D torus world
    rule.patreq = "square";

    function fredmem(oldstate,     nw, n  , ne,
                                   w, self, e,
                                   sw, s  , se) {
        var SUM_9 = nw + n + ne + w + self + e + sw + s + se;
        
        return ((oldstate << 1) & 0xFE) | (SUM_9 & 1);
    }

File: references/cellab_devkit-1.0 (1)/rules_java/fredmem.java
/*

        The Fredkin rule with the seven extra bits used as memory

*/

class Fredmem extends ruletable {

    void jcruleModes() {
        setPatternRequest("square");
    }

    int jcrule(int oldstate) {
        return ((oldstate << 1) & 0xFE) | (SUM_9 & 1);
    }
}

public class fredmem {
    public static void main(String args[]) {
        (new Fredmem()).generateRuleFile("fredmem");
    }
}
